### PR TITLE
Symintifying slice ops

### PR DESCRIPTION
**Pull request by Krovatkin**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #85196
* #85554

### CODE REVIEW

**Comment by ezyang** [comment]
Testing?

File: test/lazy/test_ts_opinfo.py [hunk 1]
```diff
@@ -175,7 +176,8 @@ def get_name(op):
         torch._lazy.mark_step()
         torch._lazy.wait_device_ops()
         prefix = "aten" if op.name in FALLBACK_LIST else "lazy"
-        found = f"{prefix}::{op.name}" in remove_suffixes(torch._lazy.metrics.counter_names())
+        symint_suffix = "_symint" if op.name in HAS_SYMINT_SUFFIX else ""
+        found = f"{prefix}::{op.name}{symint_suffix}" in remove_suffixes(torch._lazy.metrics.counter_names())
```
Comment (ezyang) [inline test/lazy/test_ts_opinfo.py:180]: This is better than https://github.com/pytorch/pytorch/pull/85264/files#diff-222fb8e63ff517f76308939550ca744de90dd6ea5706e2afd3f72db019df9b90

File: test/test_decomp.py [hunk 2]
```diff
@@ -515,6 +515,62 @@ def check_decomposed(aten_name):
...
+                            'dim': 0,
+                            'start': -10,
+                            'end': -2,
+                            'step': 2,
+                        }
+                    ),
+                ]:
+                    yield i
+        slice = SliceOpInfo()
+        self.do_cross_ref('cpu', torch.float, slice, run_all=True)
```
Comment (ezyang) [inline test/test_decomp.py:573]: Weren't you going to move this into common_methods_invocations

File: torch/_decomp/decompositions.py [hunk 3]
```diff
@@ -13,6 +14,8 @@
 from torch._prims_common.wrappers import out_wrapper
 from torch.utils._pytree import tree_flatten, tree_map
 
+old_slice = slice
```
Comment (ezyang) [inline torch/_decomp/decompositions.py:17]: this ded

File: test/test_decomp.py (hunk 2 above)
Comment (Krovatkin) [reply test/test_decomp.py:573]: it blows up spectacularly in https://github.com/pytorch/pytorch/pull/85314. I'd suggest we do it as a follow up if you don't mind. I'm usually pretty good at following up

Comment (ezyang) [reply test/test_decomp.py:573]: but... like... that the OpInfo isn't working with the other tests implies that the OpInfo is wrong...

Comment (Krovatkin) [reply test/test_decomp.py:575]: > but... like... that the OpInfo isn't working with the other tests implies that the OpInfo is wrong...

It seems it just needs way more integration work to run all the tests we run with OpInfo
* there are issues using torch.ops.aten.slice as `OpInfo.op`
* it actually passes for more types than I specified
* it also passes more forward/backward tests than I specified it should be passing.

For testing the forward of `slice` which I think was the main intent of Will's tests, it works pretty well as the part of `decomp` tests?

Comment (ezyang) [reply test/test_decomp.py:575]: I don't think it is going to be that hard to fix. I'm checking out your PR to try it.

**Comment by ezyang** [comment]
commandeering this PR

**Review by ezyang — APPROVED** [review]

File: torch/_decomp/decompositions.py [hunk 4]
```diff
@@ -609,6 +610,58 @@ def slice_backward(
     return torch.slice_scatter(grad_input, grad_output, dim, start, end, step)
 
 
+@register_decomposition(aten.slice.Tensor)
+def slice_forward(
```
Comment (bdhirsh) [inline torch/_decomp/decompositions.py:614]: dumb q: why do we need a python decomp for `slice` if we're already `c10::SymInt`-ifying the C++ implementation? Is the python impl still required for something? (either way, I totally see a python impl being useful to have).

Comment (ezyang) [reply torch/_decomp/decompositions.py:614]: Look over the patch carefully. We did not, in fact, SymInt'ify the C++ implementation.

Comment (bdhirsh) [reply torch/_decomp/decompositions.py:614]: welp, thanks. Needed to scan the codegen again to remember how it works: If we put `_symint` at the end of the kernel name in `native_functions.yaml`, then the codegen will expect you to symint-ify the C++ kernel

**Pull request closed without merging**